test/lib/mocked-database.ts

```diff
@@ -0,0 +1,68 @@
+import * as _ from 'lodash';
+import { SinonStub, stub } from 'sinon';
+import { QueryBuilder } from 'knex';
+
+import * as db from '../../src/db';
+import { Image } from '../../src/compose/images';
+
+let modelStub: SinonStub | null = null;
+// MOCKED MODELS
+const MOCKED_MODELS: Dictionary<QueryBuilder> = {};
+// MOCKED MODEL TYPES
+let MOCKED_IMAGES: Image[] = [];
+
+export function create(): SinonStub {
+	if (modelStub === null) {
+		modelStub = stub(db, 'models');
+		// Stub model requests to return our stubbed models
+		modelStub.callsFake((model: string) => {
+			return MOCKED_MODELS[model];
+		});
+	}
+	return modelStub;
+}
+
+export function restore(): void {
+	if (modelStub !== null) {
+		modelStub.restore();
+	}
+}
+
+export function setImages(images: Image[]) {
+	MOCKED_IMAGES = images;
+	return {
+		stub: stubImages,
+	};
+}
+
+function stubImages() {
```

Comment:
This unfortunately requires you to write out each object that the code you're testing will create using a QueryBuilder. If this mocked-database tool grows we will definitely look into making this more elegant.